internal/pkg/test/privilege_darwin.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package test
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"os/user"
+	"strconv"
+	"testing"
+)
+
+var origUID, origGID, unprivUID, unprivGID int
+var origHome, unprivHome string
+
+const (
+	// CacheDirPriv is the directory the cachedir gets set to when running privileged
+	CacheDirPriv = "/tmp/WithPrivilege"
+	// CacheDirUnpriv is the directory the cachedir gets set to when running unprivileged
+	CacheDirUnpriv = "/tmp/WithoutPrivilege"
+)
+
+// EnsurePrivilege ensures elevated privileges are available during a test.
+func EnsurePrivilege(t *testing.T) {
+	uid := os.Getuid()
+	if uid != 0 {
+		t.Fatal("test must be run with privilege")
+	}
+}
+
+// DropPrivilege drops privilege. Use this at the start of a test that does
+// not require elevated privileges. A matching call to ResetPrivilege must
+// occur before the test completes (a defer statement is recommended.)
+func DropPrivilege(t *testing.T) {
+	t.Fatalf("Currently not supported for Darwin")
```

Comment:
Shouldn't this be t.Skip("Currently not supported on Mac OS")?

(It's about "Skip" vs "Fatal", not "Darwin" vs "Mac OS"; I wrote "Mac OS" because I have no idea whether Mac OS users in general know what "darwin" is)